pyccel/parser/semantic.py

```diff
@@ -2682,6 +2688,32 @@ def _visit_DottedName(self, expr):
             bounding_box=(self.current_ast_node.lineno, self.current_ast_node.col_offset),
             severity='fatal')
 
+    def _visit_ListExtend(self, expr):
+        list_variable = self._visit(expr.name[0])
+        iterable = expr.name[1].args[0].value
+
+        if isinstance(iterable, (PythonList, PythonTuple)):
+            added_list = self._visit(iterable)
+            try:
+                store = [ListAppend(list_variable, a) for a in added_list]
+            except TypeError as e:
+                msg = str(e)
+                errors.report(msg, symbol=expr, severity='fatal')
+            return CodeBlock(store)
+        else:
+            pyccel_stage.set_stage('syntactic')
+            for_target = self.scope.get_new_name('index')
+            arg = FunctionCallArgument(for_target)
+            func_call = FunctionCall('append', [arg])
+            dotted = DottedName(expr.name[0], func_call)
+            lhs = PyccelSymbol('_', is_temp=True)
+            assign = Assign(lhs, dotted)
```

Comment:
@Farouk-Echaref this comment seems to have been missed. Do you have an opinion? I guess your opinion is that you prefer this code as you removed the syntactic handling? Or maybe you just hadn't seen this comment yet?

Reply:
Soory I thought I had responded to this earlier as well when I mentioned that with the current code, `ListAppend` was never called in the syntactic stage, which is why I removed that handling.

Reply:
Ok. And you think it is cleaner to do that instead of:
```suggestion
            arg = FunctionCallArgument(for_target)
            func_call = ListAppend(expr.name[0], arg)
```
?
(I'm not sure if the assign can be removed like this or not).

To be clear I don't have a strong opinion either way I would just like to know why/if you think it is better to remove the syntactic handling

Reply:
No in fact I'm working on updating this, I'll let you if I managed to update it

Reply:
Sorry, I misunderstood your question. Did you mean you wanted a method to reduce the creation of multiple objects, or just to use the `ListAppend` class instead of directly using `append`?

Reply:
I was suggesting using a syntactic `ListAppend` instead of a `DottedName` of a `FunctionCall`. Ie as shown in my example code above. Both solutions are valid I just wanted to know why you picked one over the other

Reply:
No apparent reason to be honest, my basic approach was testing the code that I wanted to generate, e.g:
```python
a = [1, 2, 3]
for i in range(start, end):
   a.append(i)
```
and from there I tried to inspect the node and construct it's content, that's why i followed the order of `FunctionCall`, `Dotted`, `Assign` ...
Im also for some reason failing to build a syntactic `ListAppend`

Reply:
If no strong preference then the easiest is to keep what you have.

I think a syntactic `ListAppend` as I find the resulting code more readable and it means that we don't have to deduce the use of `ListAppend` twice, it will be found directly with `_visit`. But if you don't have a `_visit_ListAppend` then in order to reuse the code it is probably best to do what you are doing here

Reply:
for now I'd stick with the current version, if I find the right way to use the syntactic ListAppend, i'll update it as enhancement